diff -rNu a/bundle/nginx-1.9.15/src/core/ngx_times.c b/bundle/nginx-1.9.15/src/core/ngx_times.c
--- a/bundle/nginx-1.9.15/src/core/ngx_times.c	2018-01-12 13:37:10.000000000 +0800
+++ b/bundle/nginx-1.9.15/src/core/ngx_times.c	2018-01-12 13:38:27.000000000 +0800
@@ -62,7 +62,7 @@
 void
 ngx_time_init(void)
 {
-    ngx_cached_err_log_time.len = sizeof("1970/09/28 12:00:00") - 1;
+    ngx_cached_err_log_time.len = sizeof("1970/09/28 12:00:00.000") - 1;
     ngx_cached_http_time.len = sizeof("Mon, 28 Sep 1970 06:00:00 GMT") - 1;
     ngx_cached_http_log_time.len = sizeof("28/Sep/1970:12:00:00 +0600") - 1;
     ngx_cached_http_log_iso8601.len = sizeof("1970-09-28T12:00:00+06:00") - 1;
@@ -146,10 +146,10 @@
 
     p1 = &cached_err_log_time[slot][0];
 
-    (void) ngx_sprintf(p1, "%4d/%02d/%02d %02d:%02d:%02d",
+    (void) ngx_sprintf(p1, "%4d/%02d/%02d %02d:%02d:%02d.%03d",
                        tm.ngx_tm_year, tm.ngx_tm_mon,
                        tm.ngx_tm_mday, tm.ngx_tm_hour,
-                       tm.ngx_tm_min, tm.ngx_tm_sec);
+                       tm.ngx_tm_min, tm.ngx_tm_sec, msec);
 
 
     p2 = &cached_http_log_time[slot][0];
@@ -197,6 +197,7 @@
     u_char          *p, *p2;
     ngx_tm_t         tm;
     time_t           sec;
+    ngx_uint_t       msec;
     ngx_time_t      *tp;
     struct timeval   tv;
 
@@ -207,6 +208,7 @@
     ngx_gettimeofday(&tv);
 
     sec = tv.tv_sec;
+    msec = tv.tv_usec / 1000;
 
     tp = &cached_time[slot];
 
@@ -229,10 +231,10 @@
 
     p = &cached_err_log_time[slot][0];
 
-    (void) ngx_sprintf(p, "%4d/%02d/%02d %02d:%02d:%02d",
+    (void) ngx_sprintf(p, "%4d/%02d/%02d %02d:%02d:%02d.%03d",
                        tm.ngx_tm_year, tm.ngx_tm_mon,
                        tm.ngx_tm_mday, tm.ngx_tm_hour,
-                       tm.ngx_tm_min, tm.ngx_tm_sec);
+                       tm.ngx_tm_min, tm.ngx_tm_sec, msec);
 
     p2 = &cached_syslog_time[slot][0];
 
